backend.go

```diff
@@ -111,41 +139,97 @@ func Backend() *kubeAuthBackend {
 		InitializeFunc: b.initialize,
 	}
 
-	// Set default HTTP client
-	b.httpClient = cleanhttp.DefaultPooledClient()
-
-	// Set the review factory to default to calling into the kubernetes API.
-	b.reviewFactory = tokenReviewAPIFactory
-
 	return b
 }
 
 // initialize is used to handle the state of config values just after the K8s plugin has been mounted
 func (b *kubeAuthBackend) initialize(ctx context.Context, req *logical.InitializationRequest) error {
-	// Try to load the config on initialization
-	config, err := b.loadConfig(ctx, req.Storage)
+	config, err := b.config(ctx, req.Storage)
 	if err != nil {
```

Comment:
I believe that the `InitializeFunc` will be called once after each mount operation, and its failure is for the most part ignored by the plugin runner. Your suggestion to move the run call earlier makes sense, since it ensures that a partially initialized plugin is at least being monitored properly.

Reply:
Fixed in dc905772047080ff04fc056171a0752a8d7bee76